plugins/pagerduty/src/api/client.ts

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createApiRef } from '@backstage/core';
+import {
+  Service,
+  Incident,
+  RequestOptions,
+  ClientApiConfig,
+  ServicesResponse,
+  IncidentsResponse,
+  OnCallsResponse,
+  OnCall,
+} from '../components/types';
+import { PagerDutyClient } from './types';
+
+export class UnauthorizedError extends Error {}
+
+export const pagerDutyApiRef = createApiRef<PagerDutyClient>({
+  id: 'plugin.pagerduty.api',
+  description: 'Used to fetch data from PagerDuty API',
+});
+
+export class PagerDutyClientApi implements PagerDutyClient {
+  constructor(private readonly config: ClientApiConfig) {}
+
+  async getServiceByIntegrationKey(integrationKey: string): Promise<Service[]> {
+    const params = `include[]=integrations&include[]=escalation_policies&query=${integrationKey}`;
+    const url = `${this.config.api_url}/services?${params}`;
+    const { services } = await this.getByUrl<ServicesResponse>(url);
+
+    return services;
+  }
+
+  async getIncidentsByServiceId(serviceId: string): Promise<Incident[]> {
+    const params = `service_ids[]=${serviceId}`;
+    const url = `${this.config.api_url}/incidents?${params}`;
+    const { incidents } = await this.getByUrl<IncidentsResponse>(url);
+
+    return incidents;
+  }
+
+  async getOnCallByPolicyId(policyId: string): Promise<OnCall[]> {
+    const params = `include[]=users&escalation_policy_ids[]=${policyId}`;
+    const url = `${this.config.api_url}/oncalls?${params}`;
+    const { oncalls } = await this.getByUrl<OnCallsResponse>(url);
+
+    return oncalls;
+  }
+
+  triggerAlarm(
+    integrationKey: string,
+    source: string,
+    description: string,
+    userName: string,
+  ) {
```

Comment:
Return type `Promise<Something>` perhaps? Where Something contains whatever we get from the API so we can make a link to the alert etc